### PR TITLE
Logic to calculate sol based on an input date and reference date including logging, error handling and test cases.md

### PR DESCRIPTION
**Pull request by divyalal712**

### CODE REVIEW

File: src/main/java/no/training/project/Application.java [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
+
+
     public static void main(String[] args) {
         ResourceConfig config = new ResourceConfig();
         config.register(HealthResource.class);
+        config.register(DateConversion.class);
+
 
          HttpServer httpServer = GrizzlyHttpServerFactory.createHttpServer(BASE_URI, config);
         try {
```
Comment (masoud-aghasi-td) [inline src/main/java/no/training/project/Application.java:25]: it is more natural to block the thread with `Thread.currentThread().join()` and exit with CTRL+C than waiting for any input with `System.in.read()`.

File: src/main/java/no/training/project/resources/datecalculation/DateConversion.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+            if (date == null) {
+                log.debug("Given input date is null");
+                inputDate = LocalDate.now();
+            } else {
+                log.debug("Given input date is not null");
+                inputDate = LocalDate.parse(date, DateTimeFormatter.ISO_DATE);
+                log.debug("Given input date is parsed to LocalDate");
+            }
+            LocalDate referenceDate = LocalDate.parse("2012-08-06", DateTimeFormatter.ISO_DATE);
```
Comment (masoud-aghasi-td) [inline src/main/java/no/training/project/resources/datecalculation/DateConversion.java:38]: as the referenceDate is constant at runtime its better to make it `private static final` and instantiate it once.
Also its cleaner to use a  LocalDate creator like: `LocalDate.of(2012, 8, 6);` in this case.

File: src/main/java/no/training/project/resources/datecalculation/DateConversion.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+
+            long diff = DAYS.between(referenceDate, inputDate);
+            log.info("Difference between given date reference date in days is {}", diff);
+            long sol = (long) Math.ceil(diff * 86400 / 88775.245);
```
Comment (masoud-aghasi-td) [inline src/main/java/no/training/project/resources/datecalculation/DateConversion.java:43]: its better to create constants for these literal values with meaningful names.
maybe its better to use `TimeUnit.DAYS.toSeconds(1)` than 86400.

File: src/main/java/no/training/project/resources/datecalculation/DateConversion.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            log.info("Calculated sol using thee given formula is {}", sol);
+            return sol;
+        } catch (Exception e) {
+            log.error("unexpected error while calculating sol, {}", e);
```
Comment (masoud-aghasi-td) [inline src/main/java/no/training/project/resources/datecalculation/DateConversion.java:47]: when passing exception to logger.error() method like here, it logs the exception in another manner and doesn't replace {} placeholder with any string. so the {} placeholder is unnecessary here.

**Review by masoud-aghasi-td — CHANGES_REQUESTED** [review]
Good progress! having test for edge cases is great.
I added a few clean code comments.

File: src/main/java/no/training/project/resources/datecalculation/DateConversion.java (hunk 2 above)
Comment (divyalal712) [reply src/main/java/no/training/project/resources/datecalculation/DateConversion.java:38]: Fixed this by adding 
private static final LocalDate referenceDate = LocalDate.of(2012, 8, 6);

File: src/main/java/no/training/project/resources/datecalculation/DateConversion.java (hunk 3 above)
Comment (divyalal712) [reply src/main/java/no/training/project/resources/datecalculation/DateConversion.java:43]: Fixed with:
             private static final long earthDayInSeconds = TimeUnit.DAYS.toSeconds(1);
             private static final double martianDayInSeconds = 88775.245;
and used the same in the sol calculation.

File: src/main/java/no/training/project/Application.java (hunk 1 above)
Comment (divyalal712) [reply src/main/java/no/training/project/Application.java:19]: Added Thread.currentThread().join() and exiting works with CTRL+C.

File: src/main/java/no/training/project/resources/datecalculation/DateConversion.java (hunk 3 above)
Comment (divyalal712) [reply src/main/java/no/training/project/resources/datecalculation/DateConversion.java]: Added this;
    private static final Logger LOG = LoggerFactory.getLogger(DateConversion.class);
    private static final LocalDate REFERENCE_DATE = LocalDate.of(2012, 8, 6);
    private static final long EARTHDAYINSECONDS = TimeUnit.DAYS.toSeconds(1);
    private static final double MARTIANDAYINSECONDS = 88775.245;

**Review by masoud-aghasi-td — APPROVED** [review]
